docs/proposals/26-workload-identity.md

```diff
@@ -0,0 +1,514 @@
+its name implies. In a future version of Gardener APIs, it could and would be
+nice to be renamed to `CredentialsConfig`, `CredentialsBinding`,
+`CredentialsSource` or something similar. This way it also leave open space for
+other credentials implementations in the future.
+
+Features associated with the SecretBinding like the quotas will be extended to
+also cover clusters using WorkloadIdentity for authentication.
+
+While the infrastructure credentials for the shoot cluster are the main driver
+behind this GEP, various extensions can benefit of this feature as well. For
+this purpose, a new optional field named `workloadIdentity` will be introduced
+in `shoot.spec.extensions`, it will refer to workload identity by name assuming
+the workloadIdentity resource is in the namespace of the shoot. The seed CRD
+`extensions.extensions.gardener.cloud` will be also extended to reflect that the
+extension is using workload identity letting extension controllers know when a
+JWT or other kind of credentials are used.
+
+Similar approach can be taken to provide alternative to
+`shoot.spec.dns.providers.secretName`, e.g. a new filed `workloadIdentity` will
+extended the `shoot.spec.dns.providers`.
+
+```yaml
+spec:
+  extensions:
+    - type: some-extension
+      workloadIdentity:
+        name: foo
+  dns:
+    providers:
+    - type: some-dns-provider
+      workloadIdentity:
+        name: bar
+```
+
+### Gardener as OIDC Token Issuer
+
+A new component, positively the metadata server from
+[GEP-24](./24-shoot-oidc-issuer.md), will be used to publish the public OIDC
+metadata discovery documents `/.well-known/openid-configuration` and `jwks_uri`.
+This component will be only provided with access to the public keys or any other
+public information, it will not hold or have access to any private information
+related to token generation and singing. To support keys rotation, it will serve
+also the older set of public keys so that already issued but still valid and not
+expired tokens can be used for identity federation with external services.
+
+On key rotation, the new key pair might need to be published but not used to
+sign the tokens, this is needed to ensure enough time for the external services
+to discover the new public key. This rotation strategy could be useful for
+external services that do not automatically rediscover the OIDC issuer metadata
+when the token is signed with still unknown to them key.
+
+The Kubernetes API server extended by the Gardener API server is already issuing
+JWTs for the kubernetes service accounts. To completely separate workload
+identity JWTs from service accounts JWTs, Gardener API will accept issuer URL
+parameter which value should not be the same as the issuer of the kubernetes
+service accounts. The workload identity issuer url should not be among the
+accepted issuers of the Kubernetes API server. Other configuration options for
+the Gardener API server will be the private key used to sign the tokens, the
```

Comment:
Can you augment the document to also discuss the aspects of how this is envisioned to be integrated into `gardener-operator`? It deploys GAPI, so it will also be responsible for setting this private key. Will it be auto-generated and rotated via the typical credentials rotation operations (after human operators add respective operation annotation)?